app/controllers/holds_controller.rb

```diff
@@ -42,10 +43,14 @@ def batch_update
   #
   # GET /holds/new
   def new
+    raise NewHoldException, 'Error' if params[:catkey].blank?
+
     form_builder = PlaceHoldForm::Builder.new(catkey: params[:catkey],
                                               user_token: current_user.session_token,
                                               client: symphony_client)
     @place_hold_form_params = form_builder.generate
+
+    raise NewHoldException, 'Error' if @place_hold_form_params.blank?
```

Comment:
What is the scenario where this raises that the raise on line 46 doesn't catch?

Reply:
the first one is for when there is no catkey passed to the new form like by just going to `/holds/new` 
the other one is when the record is not holdable, form_builder will return nil

Reply:
right! :) thanks